### PR TITLE
Implement selection panel add group notice

### PR DESCRIPTION
**Pull request by nfmohit**
## Summary

Addresses issue:

- #8159 

## Relevant technical choices

This PR implements an add group notice to the Audience Selection Panel when there is only one group selected.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

**Review by techanvil — CHANGES_REQUESTED** [review]
Hi @nfmohit, nice work here so far. However - it doesn't fully meet the AC. I realise the IB itself doesn't fully meet the AC as originally specced, so no worries, these things happen!

Anyhow, the notice should only be displayed if there is just one audience selected when the panel is initially opened, rather than any time the audience selection count in the panel is reduced to one.

The idea is that we want to draw the user's attention to this if they have a saved selection of just one audience, but not nag them too much while they are simply interacting with the selection.

One a related note - I realise this aspect wasn't made clear at all but once a user has selected another audience, the notice should disappear and should not show up again while the panel is still open even if the user reduces the selection to a single audience. I've amended the AC to make this point clear, sorry this was a last minute addition.

**Comment by nfmohit** [comment]
Thank you for the clarification, @techanvil! I've updated the PR accordingly. Please let me know if this looks good now.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/index.test.js [hunk 1]
```diff
@@ -275,6 +281,103 @@ describe( 'AudienceSelectionPanel', () => {
...
+			await waitForRegistry();
+
+			expect(
+				getByText(
+					/By adding another group to your dashboard, you will be able to compare them and understand which content brings back users from each group/i
+				)
+			).toBeInTheDocument();
+		} );
+
+		it( 'should not display notice when there is a saved selection of less than or more than one group', async () => {
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/index.test.js:306]: This describes the case where there is less than one group in the saved selection, but doesn't test it.

Please either add test coverage for the zero-selection case (maybe splitting the test into two or using `it.each()`), or amend the description.

**Review by techanvil — COMMENTED** [review]
Thanks @nfmohit!

I've left a comment on one of the tests, please take look.

Beyond that - I must admit that when testing it, the behaviour does feel slightly odd. Notably the case where the user opens the selection panel with a single saved audiences, and then _unchecks_ the audience.

At this point, the notice disappears, but the user hasn't actually given an implicit indication of having taken action in accordance with the notice - they've removed their audience rather than adding one as the notice suggests.

Anyhow, let's not worry about that here - I will raise a separate issue to refine the behaviour to address this.

**Review by techanvil — CHANGES_REQUESTED** [review]
Whoops, forgot to request changes on my previous review!

**Comment by techanvil** [comment]
> Beyond that - I must admit that when testing it, the behaviour does feel slightly odd. Notably the case where the user opens the selection panel with a single saved audience, and then _unchecks_ the audience.
> 
> At this point, the notice disappears, but the user hasn't actually given an implicit indication of having taken action in accordance with the notice - they've removed their audience rather than adding one as the notice suggests.
> 
> Anyhow, let's not worry about that here - I will raise a separate issue to refine the behaviour to address this.

Just noting I've added https://github.com/google/site-kit-wp/issues/8909 to address the above.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/index.test.js (hunk 1 above)
Comment (nfmohit) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/index.test.js]: I've updated the test coverage using `it.each()` as recommended, thanks Tom!

**Comment by nfmohit** [comment]
I agree, thank you for covering this behaviour. I've addressed the CR feedback, thanks @techanvil !

**Review by techanvil — APPROVED** [review]
LGTM, nice one @nfmohit!
